blob.py

```diff
@@ -65,25 +68,49 @@ def distanceToTarget(self, target):
     def eat(self, food):
         food.removeFromWorld()
         self.food += 1
+        self.energy += 2
 
     # Method for day ending
     def endDay(self, day, world):
-        if (self.food > 2):
+        if (self.energy > 2):
             self.replicate(day)
+            self.energy -= 2
         elif (self.food == 0):
             self.kill(day, world)
 
     # Method returning direction to target in degrees
     def getDirectionToTarget(self, target):
-        positionFromTarget = (target[1] - self.position[1], target[0]- self.position[0])
+        positionFromTarget = (
+            target[1] - self.position[1], target[0] - self.position[0])
         directionToTarget = degrees(atan2(*positionFromTarget))
         return directionToTarget
 
     # Method returning blob position (in cells)
     def getPosition(self):
         return self.position
 
+    # Method returning blob remaining energy
+    def getEnergy(self):
+        return self.energy
+
+    # Method useEnergy if blob is not in home, replicate if energy more then 2
+    def useEnergy(self, day, world):
+        if (self.inHome() == False):
+            self.energy = self.getEnergy(
+            ) - (self.relativePosition[1] - self.relativePosition[0])
```

Comment:
Completely unclear, better make it one-liner. It's not a tuple or anything, simple subtraction.